go/extractor/extractor.go

```diff
@@ -1561,6 +1562,13 @@ func extractType(tw *trap.Writer, tp types.Type) trap.Label {
 				extractMethod(tw, meth)
 
 				extractComponentType(tw, lbl, i, meth.Name(), meth.Type())
+
+				// meth.Id() will be equal to meth.Name() for an exported method, or
+				// packge-qualified otherwise.
```

Comment:
Question: My understanding is that, in Go, exported methods always start with an upper-case character. Why did you go for this `meth.Id() != meth.Name()` check rather than checking the first character of the name? Did that not work or do you think this method is more reliable or has other advantages?